python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -119,60 +122,51 @@ def categorical_embedding_with_indices(feature_tensor, feature_info, file_io: Fi
-    """
+    Args under feature_layer_info:
+        vocabulary_file: string; path to vocabulary CSV file for the input tensor containing the vocabulary to look-up.
+                        uses the "key" named column as vocabulary of the 1st column if no "key" column present.
+        max_length: int; max number of rows to consider from the vocabulary file.
+                        if null, considers the entire file vocabulary.
+        embedding_size: int; dimension size of the embedding;
+                        if null, then the tensor is just converted to its one-hot representation
+        encoding_size: int: dimension size of the sequence encoding computed using a biLSTM
 
-    def __init__(self, vocabulary_keys, vocabulary_ids, num_oov_buckets, feature_name):
-        super(VocabLookup, self).__init__(trainable=False, dtype=tf.int64)
-        self.vocabulary_keys = vocabulary_keys
-        self.vocabulary_ids = vocabulary_ids
-        self.vocabulary_size = len(set(vocabulary_ids))
-        self.num_oov_buckets = num_oov_buckets
-        self.feature_name = feature_name
+    NOTE:
+        The input dimension for the embedding is fixed to 256 because the string is
+        converted into a bytes sequence.
+    """
+    args = feature_info.get("feature_layer_info")["args"]
 
-    def build(self, input_shape):
-        table_init = lookup.KeyValueTensorInitializer(
-            keys=self.vocabulary_keys,
-            values=self.vocabulary_ids,
-            key_dtype=tf.string,
-            value_dtype=tf.int64,
-        )
-        self.lookup_table = lookup.StaticVocabularyTable(
-            initializer=table_init,
-            num_oov_buckets=self.num_oov_buckets,
-            name="{}_lookup_table".format(self.feature_name),
-        )
-        self.built = True
+    categorical_indices, vocabulary_keys, num_oov_buckets = get_categorical_indices_from_vocabulary(
+        feature_info, feature_tensor, file_io
+    )
 
-    def call(self, input_text):
-        return self.lookup_table.lookup(input_text)
+    vocabulary_size = len(set(vocabulary_keys))
+    categorical_embeddings = layers.Embedding(
+        input_dim=vocabulary_size + num_oov_buckets,
+        output_dim=args["embedding_size"],
+        mask_zero=True,
+        input_length=args.get("max_length")
+    )(categorical_indices)
 
-    def get_config(self):
-        config = super(VocabLookup, self).get_config()
-        config.update(
-            {
-                "vocabulary_keys": self.vocabulary_keys,
-                "vocabulary_ids": self.vocabulary_ids,
-                "vocabulary_size": self.vocabulary_size,
-                "num_oov_buckets": self.num_oov_buckets,
-                "feature_name": self.feature_name,
-            }
-        )
-        return config
+    categorical_embeddings = tf.squeeze(categorical_embeddings, axis=1)
+    encoding = get_bilstm_encoding(categorical_embeddings, int(args["encoding_size"] / 2))
+    return encoding
```

Comment:
@Ullimague Can we move this to a separate function as well under `sequence.py` perhaps? The reason is I want it to be reused for `bytes_sequence_to_encoding_bilstm` instead of defining the same in two places. I like the refactoring you've done until now. Good place for ml4ir.